shell/platform/android/test/io/flutter/plugin/platform/PlatformViewsControllerTest.java

```diff
@@ -399,11 +400,44 @@ public void itUsesActionEventTypeFromMotionEventForHybridPlatformViews() {
             original.getFlags(),
             motionEventId.getId());
 
-    MotionEvent resolvedEvent =
+    MotionEvent resolvedNonVdEvent =
         platformViewsController.toMotionEvent(
-            /*density=*/ 1, frameWorkTouch, /*usingVirtualDisplay=*/ false);
+            1, // density
+            frameWorkTouchNonVd,
+            false // usingVirtualDisplays
+            );
 
-    assertEquals(resolvedEvent.getAction(), frameWorkTouch.action);
+    // Re track the original motion event, as toMotionEvent will pop it from the motionEventTracker.
+    motionEventId = motionEventTracker.track(original);
+    PlatformViewTouch frameWorkTouchVd =
+        new PlatformViewTouch(
+            0, // viewId
```

Comment:
if this is the same code as above, could you factor it into a helper function to avoid the code duplication?

Reply:
done